services/s3/src/main/java/software/amazon/awssdk/services/s3/internal/crt/DefaultS3CrtAsyncClient.java

```diff
@@ -110,44 +111,17 @@ private static S3CrtAsyncHttpClient.Builder initializeS3CrtAsyncHttpClient(Defau
 
     public static final class DefaultS3CrtClientBuilder implements S3CrtAsyncClientBuilder {
         private Long readBufferSizeInBytes;
-        private AwsCredentialsProvider credentialsProvider;
+        private IdentityProvider<? extends AwsCredentialsIdentity> credentialsProvider;
         private Region region;
         private Long minimalPartSizeInBytes;
         private Double targetThroughputInGbps;
         private Integer maxConcurrency;
         private URI endpointOverride;
         private Boolean checksumValidationEnabled;
 
-        public AwsCredentialsProvider credentialsProvider() {
-            return credentialsProvider;
-        }
-
-        public Region region() {
-            return region;
-        }
-
-        public Long minimumPartSizeInBytes() {
-            return minimalPartSizeInBytes;
-        }
-
-        public Double targetThroughputInGbps() {
-            return targetThroughputInGbps;
-        }
-
-        public Integer maxConcurrency() {
-            return maxConcurrency;
-        }
-
-        public URI endpointOverride() {
-            return endpointOverride;
-        }
-
-        public Long readBufferSizeInBytes() {
-            return readBufferSizeInBytes;
-        }
-
```

Comment:
Why are these methods removed?

Reply:
They are unused methods. The constructor logic of the containing class DefaultS3CrtAsyncClient accesses the private memebers of the builder directly (e.g., `builder.region`). And DefaultS3CrtAsyncClient is `@SdkInternalApi` so thought it is safe to remove them. LMK if I shouldn't.

Reply:
🤦🏼‍♀️ I don't know why I added them in the first place(maybe I just accidentally generated them on Intellij) Agreed that we should remove them, but we probably should not do it as part of this PR in the spirit of keeping PRs small.

Reply:
I needed to change the return type of the `credentialsProvider()` for this PR to the new type, but this method was unused, so removed it, along with the other unused ones. I can move the removals to a separate PR.

Reply:
Moved to separate PR https://github.com/aws/aws-sdk-java-v2/pull/3879